test/e2e/alphaupdate/update_test.go

```diff
@@ -295,3 +350,40 @@ func validateCustomCodePreservation(projectDir string) {
 	Expect(err).NotTo(HaveOccurred())
 	Expect(string(content)).To(ContainSubstring(controllerImplementation))
 }
+
+func validateConflictMarkers(projectDir string) {
+	By("validating conflict markers are present")
+	filesToCheck := []string{
+		filepath.Join(projectDir, "api", "v1", "testoperator_types.go"),
+		filepath.Join(projectDir, "internal", "controller", "testoperator_controller.go"),
+	}
+
+	conflictMarkersFound := false
+	for _, file := range filesToCheck {
+		content, err := os.ReadFile(file)
+		if err != nil {
+			continue
+		}
+		fileContent := string(content)
+		if strings.Contains(fileContent, "<<<<<<<") && strings.Contains(fileContent, "=======") && strings.Contains(fileContent, ">>>>>>>") {
+			conflictMarkersFound = true
+			break
+		}
+	}
+	Expect(conflictMarkersFound).To(BeTrue(), "Expected to find conflict markers in at least one file")
+}
+
+func validateConflictState(projectDir string) {
+	By("validating merge stopped with conflicts requiring manual resolution")
+	
+	// Check that working directory has uncommitted changes (conflict state)
+	cmd := exec.Command("git", "status", "--porcelain")
+	cmd.Dir = projectDir
+	output, err := cmd.CombinedOutput()
+	Expect(err).NotTo(HaveOccurred())
+	statusOutput := strings.TrimSpace(string(output))
+	Expect(statusOutput).NotTo(BeEmpty(), "Working directory should have uncommitted changes from merge conflict")
+
+	// Reuse existing conflict marker validation
```

Comment:
```suggestion
```
This comment is unnecessary
When we write code, it needs to be readable in a way that we do not need comments
Like reading a history, we are able to understand what is happening in each step with function names, etc.